src/api/jaeger.js

```diff
@@ -78,8 +79,11 @@ const JaegerAPI = {
   fetchTrace(id) {
     return getJSON(`${this.apiRoot}traces/${id}`);
   },
+  archiveTrace(id) {
```

Comment:
How do you find out if a trace is archived already? Is it in the trace response from `GET /trace/:traceId` ?

Reply:
Basically, you can't. International trace of mystery. Not ideal, and in #7 we discuss that.

Reply:
@yurishkuro IMO, I think this should be a requirement of the API before this is implemented.
 
When the user refreshes the page, you won't know the trace has been archived. It will confuse the user when they click the button and its toggled to "Archived", then refresh the page and its no longer toggled.

Reply:
I think perfect is the enemy of good. I'd rather have this feature in prod helping users than block on implementing the additional APIs.

The act of archiving already has pretty active notifications, so unlikely to confuse the user who's doing the archiving.

Reply:
Let's leave it undocumented until it's fully implemented.

Reply:
Maybe even explain to the user when they click the button or hover over with a tooltip.

Reply:
I was thinking that too. Not sure what to say, though. 

> This trace has been archived. But, if you refresh this page, the archive button will still be enabled. For now (unfortunately) we don't disable the button when you have already archived the trace. This feature is a WIP, you can follow the implementation at #7. Please contact @yurishkuro with questions or to verify the trace has been archived.

Ok, not that last part, but the other stuff...? Kind of verbose, but not sure how to put it succinctly.

Reply:
SGTM.

Reply:
@yurishkuro What do you think of adding this to the success notification?

>This trace has been archived. But, if you refresh this page, the archive button will still be enabled. For now (unfortunately) we don't disable the button when you have already archived the trace. This feature is a WIP, you can follow the implementation at #7. [Please contact @yurishkuro with questions or to verify the trace has been archived.]

Exact phrasing on that last part is TBD...

Reply:
> What do you think of adding this to the success notification?

I think we should not do that. It should only say "This trace has been archived." I think you guys are arguing for a feature that (a) nobody asked for yet, and (b) costs way more to implement than its benefits. If someone needs this feature in the future, let them open a ticket, get the votes, and get someone to prioritize that above all the other work & improvements we could be spending time on.

Reply:
@yurishkuro I'm not sure which feature you're referring to, when you say nobody asked for it - but I'd say that displaying the archived status is an integral part of the archive feature. Shipping this without displaying the status or at least providing messaging feels half-baked to me.

Reply:
we ARE providing a message on successful save.

Reply:
Confirmation of the save is one thing, it's another provide feedback on an ongoing basis. "Is this trace going to disappear tomorrow or not?" 

I disagree the costs outweigh the benefits - UX is important. Certainly less so when this is an undocumented feature (this was some context I was missing) but still important. I personally think the UX needs more work before the feature is released to the public but the technical implementation looks good to me so the PR has a thumbs up from me.